### ISSUE TITLE
TMDB TV scraper overrides any TVDB scraper on manual import

### ISSUE THREAD

**Issue by maffo999** [xbmc/xbmc#25743]
### Describe the bug
On my Kodi 21 installation, I have both TMDB TV scraper (installed by default) and TVDB scraper (installed from repository and preferred to the default one).
I have added my TV shows library and told Kodi to scrape it using TVDB, however I noticed one show did not get a proper match.
I removed the show from the library, and choose "Scan to library" to manually scrape the show.
I was presented with a choice of multiple possibilities and chose the right one , but once scraping was completed I noticed the 2nd season was completely missing.
After trying again the steps above while looking in the logs, I notice that despite telling Kodi the TV library should be scraped with TVDB, any manual "Scan to library" will instead be done with TMDB TV

## Expected Behavior
Kodi should respect the scraper chosen for the library also when performing manual "Scan to library"

## Actual Behavior
Kodi does not respect the scraper chosen for the library and defaults to using TMDB TV

### To Reproduce
1. Perform a clean installation of Kodi 21
2. Install TVDB video information provider
3. Have any show show present in your media library 
4. Add the TV shows folder to Kodi library selecting TVDB video scraper
5. The show will be matched to a wrong match
6. Remove the show from the library
7. Enable debug log
8. Go to the file manager and select the folder containing the show
9. Right click and select "Scan to library"
10. From the choices presented, choose the right TV show
11. Watch the logs to see Kodi using TMDB TV for scraping (instead of TVDB configured on the library)

### Debuglog
The debuglog can be found here:
https://paste.kodi.tv/ojibuhizix.kodi

### Screenshots 
Unable to take screenshots at the moment since Kodi is connected to the TV directly, if needed I can provide a picture confirming TVDB is the scraper selected for the TV library

## Additional context or screenshots (if appropriate)
TVDB usually has more updated info on TV shows, for example "Monster - The Jeffrey Dahmer story" already has season 2 information - compared to TMDB which only has season 1 info

### Your Environment
Used Operating system:
Linux kodi 6.6.47+rpt-rpi-2712 #1 SMP PREEMPT Debian 1:6.6.47-1+rpt1 (2024-09-02) aarch64 GNU/Linux
PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
Kodi Media Center 21.1 (21.1.0) Git:20240830-39fb0e3db7

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by maffo999** [comment]
I have found what I believe is a workaround for this - at least for the moment:
1) delete the show from the library
2) clean library from the settings menu
3) rescan the whole library instead of using "Scan to library" for the selected show

By doing this, the show is now scraping correctly with all seasons by taking data from TVDB

**Comment by KarellenX** [comment]
In future, please provide the full log. There was not enough info there to follow the complete process. But it certainly looks like the problem was on your setup. 

I have tested this and could not replicate the problem.

Looking at line 8 `debug <general>: CVideoLibraryRefreshingJob: user selected item 'Monster (2022)' with URL '271289'`

**271289** does not exist at TMDB, and points to a completely different show at TVDB. No idea where this ID has come from. NFO file? Previous bad scrape? Selecting the wrong show when prompted?

Deleting the show from the library will have cleared out the incorrect episodeguide and allowed the correct show to be scraped.

**Comment by maffo999** [comment]
@KarellenX it was a brand new setup and the log was huge so I tried to trim it to the relevant part. 
The .nfo file for the show contains the correct entries:
```
<tvshow>
  <title>Monster (2022)</title>
  <plot>Anthology series focusing on the lives and crimes of serial killers.</plot>
  <mpaa>18</mpaa>
  <id>389492</id>
  <uniqueid type="tvdb" default="true">389492</uniqueid>
  <uniqueid type="imdb">tt24513650</uniqueid>
  <uniqueid type="tmdb">113988</uniqueid>
  <uniqueid type="tvmaze">50907</uniqueid>
```

Since I should still be able to reproduce the problem, could you please tell me from what moment should I collect the log?
It seems to me that for the issue I reported (TVDB scraper being ignored and TMDB TV being used) it's clearly visible how Kodi is starting the incorrect addon:
```
2024-09-19 21:48:55.363 T:908     debug <general>: ------ Window Deinit (DialogSelect.xml) ------
2024-09-19 21:48:55.378 T:908     debug <general>: CVideoLibraryRefreshingJob: user selected item 'Monster (2022)' with URL '271289'
2024-09-19 21:48:55.378 T:908     debug <general>: DialogProgress::Open called 
2024-09-19 21:48:55.378 T:908     debug <general>: ------ Window Init (DialogConfirm.xml) ------
2024-09-19 21:48:55.828 T:908     debug <general>: VideoInfoScanner: Fetching url '271289' using TMDb TV Shows scraper (content: 'tvshows')
```

**Comment by KarellenX** [comment]
Yes, I did gloss over your reported problem - TMDB being used instead of TVDB. That is because I did not have the full log, and I would also have needed the database for that particular problem, neither of which was provided. But I did test this three times and could not replicate the issue. Also, using Scan to Library is normal behaviour for my own library (I never run a full Library Update) and have never experienced problems with it.

Also the spurious 271289 makes me believe you had some sort of "junk" somewhere either in the database or  in the nfo file, and again, you have not provided the full nfo file. Please stop "trimming" things to what you _think_ I want. I need full files.

271289 - Does not exist at TMDB- https://www.themoviedb.org/tv/271289-monster
Digging a bit deeper, looks like it was deleted as a dupicate. So you were unlucky enough to scrape when it was available, but rescraping for new episodes caused it to fail as it no longer existed.

271289 - Does not exist at TVDB either.

A Refresh of the TV Show (not the episodes) and ignoring the nfo file will have cleared out the old data and downloaded the new ID's for the show and solved the scraping problem- but not the issue of wrong scraper.

If you still think there is a problem, ensure that you can still replicate the problem and provide the following:
1. Full nfo file - though I already know you have old ID's in there due to the show being deleted
2. MyVideos database from your Userdata folder
3. Full log that captures you scraping this show (which uses the wrong scraper) and another show that does use the correct TVDB scraper using the Scan to Library method.

And I would prefer to continue this on the forum. Do you have a forum account?

**Comment by maffo999** [comment]
@KarellenX I do not have a forum account, but I have an account here so that's why I reported it here.

At what point in time do you need me to collect the MyVideos database?

> Full log that captures you scraping this show (which uses the wrong scraper) and another show that does use the correct TVDB scraper using the Scan to Library method.

This will not be possible. Every time "Scan to library" is used, TMDB TV is used for scraping instead of TVDB.

**Comment by KarellenX** [comment]
>At what point in time do you need me to collect the MyVideos database and where do you want it to be uploaded?

Upload it to a file hosting site and provide the link so I can download it (on the forum, you could have sent me the link privately). Just grab the DB when you are ready to upload. Nothing special needs to be done to it.

>This will not be possible. Every time "Scan to library" is used, TMDB TV is used for scraping instead of TVDB.

Ok, provide the full log that captures that. Ignore my TVDB request. Make sure the log is captured from startup to shutdown including the scan. Post it to our Kodi paste site. Same with the nfo file as I want to find where that ID is coming from.

Thanks.

**Comment by maffo999** [comment]
Here's the full Kodi logfile from the moment I started the system: https://paste.kodi.tv/takinipeyi.kodi
Order of actions:
1) Started Kodi
2) Enabled debug logging
3) Performed "Scan to library" on show folder

Here's the `.nfo` file in the folder of the show: https://paste.kodi.tv/ecoteliqab

Here's the database: https://file.io/SI2ulTzA6kTl

**Comment by KarellenX** [comment]
Thanks. Log looks ok. I'll check that in detail when I have the database. The database link does not work- says it has been deleted. Can you try again?

The tvshow.nfo file is minimal. How did you create it? Media Manager? Ah, I see. You are using Sonarr. I strongly urge you not to use nfo files from Sonarr as they are not fully compatible.

You also have nfo files for all the episodes, including s02. Kodi is reading those nfo files, so it really does not matter what scraper you have active as it is not being used. It is being called, but not used.

**Comment by maffo999** [comment]
@KarellenX  it's not easy to find a public upload site which will accept sqlite db files without issues, I have tried to share it from my private NextCloud instance and hope this works: https://maffo.hopto.org/s/edtc9DcF5wxwSN9

>Ah, I see. You are using Sonarr. I strongly urge you not to use nfo files from Sonarr as they are not fully compatible.

>You also have nfo files for all the episodes, including s02. Kodi is reading those nfo files, so it really does not matter what scraper you have active as it is not being used. It is being called, but not used.

I was not aware Kodi would always favor local `.nfo` files mainly because I never keep `.nfo` files, but I got Sonarr creating them for this show assuming it might help Kodi find the correct information; however the issue (TMDB over TVDB) is present regardless of the `.nfo` file existence.
If you want I can remove all `.nfo` files from this show and reproduce it again.

**Comment by KarellenX** [comment]
Ok, received the database and you can delete the link from your post.

Do you have the ability to open the database? If not, install this... https://sqlitebrowser.org/

1. Open the database
2. Go to the **path** table
3. sort the columns on **strScraper** descending
4. You will see three folders listed at the top with corresponding scrapers linked to them. The first and last are your tv and movie sources. The third is the problem show Monster. Somehow you have specifically set TheMovieDB scraper for that show.

In Files>View, go to your Source folder then to Monster (2022) folder and change the scraper to TVDB.


>I was not aware Kodi would always favor local .nfo

Yes, local data has priority- nfo and artwork. If local is not found, then scrapers are called.

**Comment by maffo999** [comment]
@KarellenX I'm sorry but I feel there's some communication breakdown. 
I was already able to fix the problem myself with the workaround mentioned in my first comment to this post. I also understand there was some backend data issues since automatic scraping was matching this show to an Anime but that was fixed between the time I opened this thread and posted the first comment with the workaround found.
I apologize English is not my first language so I will try to expose the problem once again:
**When a TV show library is configured with TVDB scraper and a show is manually added via "Scan to library", Kodi will use TMDB TV as source (default and cannot be disabled) instead of the scraper configured on the library.**

>     Open the database
>     Go to the path table
>     sort the columns on strScraper descending
>     You will see three folders listed at the top with corresponding scrapers linked to them. The first and last are your tv and movie sources. The third is the problem show Monster. Somehow you have specifically set TheMovieDB scraper for that show.
> 
> In Files>View, go to your Source folder then to Monster (2022) folder and change the scraper to TVDB.

I have done that and as you can see the scraper for that show was TMDB while for the whole library is TVDB. I have never chosen TMDB as scraper for my library so I would like to understand why TMDB is being used instead of the one I choose.

**Comment by KarellenX** [comment]
No, there is no communication breakdown. As I have repeated numerous times, I understand you are stating that TMDB is being used instead of TVDB. But I am starting to believe that you may be ignoring my explanation.

There is no Kodi bug. Rather it is user error.

The reason for that, and all evidence points to this, is that you have forced TMDB as the default scraper **for that particular show**. I assume you have done this accidentally. It is possible to, not only set a default scraper for an entire Source, but to also set scrapers for individual TV Shows and Movies. This is what you have done without realising it. 

As you have already stated that you have solved the problem, I now consider this matter closed, and don't really want to be drawn into repetitive explanations.

If you have any other issues, please create a forum account and post your question there.

Regards

**Comment by maffo999** [comment]
I honestly don't know anymore how to try and convince you that TMDB was not set to be used anywhere and also I don't understand why it was used in the first place, but I understand you were not able to reproduce it and user error is the easiest way out.
Thanks for looking into this nevertheless.
